internal/backend/kubernetes/config.go

```diff
@@ -4,52 +4,27 @@ import (
 	"context"
 
 	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
-	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
-// Config used by the NewBackend function family.
-type Config struct {
-	// Config is the Kubernetes client configuration.
-	*rest.Config
-
-	// Namespace restricts the scope of the backend such that Hardware objects are retrieved from
-	// this namespace only. Defaults to "default".
-	Namespace string
-
+// BackendConfig used by the NewBackend function family.
+type BackendConfig struct {
```

Comment:
This configuration object has been simplified such that it can be passed straight to `NewBackend`. `NewBackend` now constructs the rest client provided `Config.ClientConfig` is unspecified.